crates/astria-sequencer-relayer/src/relayer/read.rs

```diff
@@ -185,6 +185,9 @@ async fn fetch_block(
         .max_delay(block_time)
         .on_retry(
             |attempt: u32, next_delay: Option<Duration>, error: &eyre::Report| {
+                metrics::counter!(crate::metrics_init::SEQUENCER_BLOCK_FETCH_FAILURE_COUNT)
```

Comment:
This counter only makes sense in the context of the retry future since the fetch fails hard (after `u32::MAX` retries).

Reply:
Agreed an additional guage would make sense here.